Realm/RLMManagedSet.mm

```diff
@@ -0,0 +1,572 @@
+static void throwError(__unsafe_unretained RLMManagedSet *const ar, NSString *aggregateMethod) {
+    try {
+        throw;
+    }
+    catch (realm::InvalidTransactionException const&) {
+        @throw RLMException(@"Cannot modify managed RLMSet outside of a write transaction.");
+    }
+    catch (realm::IncorrectThreadException const&) {
+        @throw RLMException(@"Realm accessed from incorrect thread.");
+    }
+    catch (realm::object_store::Set::InvalidatedException const&) {
+        @throw RLMException(@"RLMSet has been invalidated or the containing object has been deleted.");
+    }
+    catch (realm::object_store::Set::InvalidEmbeddedOperationException const&) {
+        @throw RLMException(@"Cannot add an embedded object to an RLMSet.");
+    }
+    catch (realm::Results::UnsupportedColumnTypeException const& e) {
+        if (ar->_backingSet.get_type() == realm::PropertyType::Object) {
+            @throw RLMException(@"%@: is not supported for %s%s property '%s'.",
+                                aggregateMethod,
+                                string_for_property_type(e.property_type),
+                                is_nullable(e.property_type) ? "?" : "",
+                                e.column_name.data());
+        }
+        @throw RLMException(@"%@: is not supported for %s%s set '%@.%@'.",
+                            aggregateMethod,
+                            string_for_property_type(e.property_type),
+                            is_nullable(e.property_type) ? "?" : "",
+                            ar->_ownerInfo->rlmObjectSchema.className, ar->_key);
+    }
+    catch (std::logic_error const& e) {
+        @throw RLMException(e);
+    }
+}
+
+template<typename Function>
+static auto translateErrors(__unsafe_unretained RLMManagedSet *const set,
+                            Function&& f, NSString *aggregateMethod=nil) {
+    try {
+        return f();
+    }
+    catch (...) {
+        throwError(set, aggregateMethod);
+    }
+}
+
+template<typename Function>
+static auto translateErrors(Function&& f) {
+    try {
+        return f();
+    }
+    catch (...) {
+        throwError(nil, nil);
+    }
+}
+
+static void changeSet(__unsafe_unretained RLMManagedSet *const set,
+                      dispatch_block_t f) {
+    translateErrors([&] { set->_backingSet.verify_in_transaction(); });
+
+    RLMObservationTracker tracker(set->_realm, false);
+    tracker.trackDeletions();
+    auto obsInfo = RLMGetObservationInfo(set->_observationInfo.get(),
+                                         set->_backingSet.get_parent_object_key(),
+                                         *set->_ownerInfo);
+    if (obsInfo) {
+        tracker.willChange(obsInfo, set->_key);
+    }
+
+    translateErrors(f);
+}
+
+//
+// public method implementations
+//
+- (RLMRealm *)realm {
+    return _realm;
+}
+
+- (NSUInteger)count {
+    return translateErrors([&] { return _backingSet.size(); });
+}
+
+- (NSArray<id> *)allObjects {
```

Comment:
Shouldn't this be an NSSet?

Reply:
This is mirroring the NSSet API which returns an NSArray of all items in the set.